### ISSUE TITLE
bug(callers,callees): "index empty" hint fires when 157K edges exist (built-marker false-negative cascades into misleading agent next_step)

### ISSUE THREAD

**Issue by aimasteracc** [aimasteracc/tree-sitter-analyzer#981]
From dogfood session on `develop` @ 35d0dac0 (2026-06-16).

**Symptom (agent-facing, critical)**

`--callers` / `--callees` on a symbol that does not exist returns:

```json
{
  "success": true,
  "verdict": "NOT_FOUND",
  "caller_count": 0,
  "next_step": "Call-graph index is empty or has not been built yet. Run `tree-sitter-analyzer --full-index` first, then retry."
}
```

…but the index is **fully built**:

```
$ TSA --ast-cache --ast-cache-mode stats
indexed_files: 1955
edges_by_kind: {calls: 125810, contains: 21833, extends: 728, imports: 9136}
```

**Repro**

```
uv run python -m tree_sitter_analyzer --callers "nonexistent_xyz_abc" --format json
uv run python -m tree_sitter_analyzer --callees "nonexistent_xyz_abc" --format json
```

Both emit the misleading "index empty" hint. Meanwhile the sibling tool gets it right:

```
uv run python -m tree_sitter_analyzer --codegraph-navigate "nonexistent_xyz" --format json
# → "Symbol 'nonexistent_xyz' not in the index. Check spelling or run index action=auto to rebuild."
```

**Root cause**

`callers_tool.py:210` and `callees_tool.py` gate the "index empty" hint on `has_any_call_edges = call_graph_built` (the **built marker**), not on actual edge count:

```python
# callers_tool.py:147-166
call_graph_built = self._cache_call_graph_built(cache) if cache else False
...
has_any_call_edges = call_graph_built   # ← marker, not data
...
if result.get("verdict") == "NOT_FOUND" and not has_any_call_edges:
    index_hint = "Call-graph index is empty or has not been built yet..."
```

Direct probe:

```
$ python -c "import sqlite3; from tree_sitter_analyzer._ast_cache_callgraph_state import call_graph_built; print(call_graph_built(sqlite3.connect('.ast-cache/index.db')))"
False
```

So the built marker is in a **false-negative state** (the same class of bug as open #978), and `callers`/`callees` blindly trust it instead of consulting `cache.has_call_edges()` / the actual `edges` table count (157,507 rows). The navigate tool does not have this dependency and is correct.

**Why this matters for agents**

1. The tool actively lies about index state — every `NOT_FOUND` is framed as "your index is missing" rather than "this symbol genuinely does not exist".
2. An agent following the hint runs `--full-index` (slow). Per #978, a mixed incremental `index_project()` may **re-stamp the marker false-negative again**, so the agent enters a re-index loop with no escape.
3. `success: true` paired with `verdict: NOT_FOUND` is internally contradictory — agents that gate on `success` will treat a failed lookup as a successful empty result.

**Fix direction (two layers)**

A. **Consumer-side (this issue):** in `callers_tool.py` / `callees_tool.py`, change the hint gate to consult actual edges, not just the marker. Sketch:

```python
has_any_call_edges = call_graph_built or (cache is not None and cache.has_call_edges())
```

And split the message:

- if `cache.has_call_edges()` is True but symbol missing → `f"Symbol {func_name!r} not in the index. Check spelling or run --codegraph-navigate to browse."`  (port the navigate phrasing)
- if `cache.has_call_edges()` is False → keep the existing "index empty, run --full-index" hint
- set `success: false` when verdict is `NOT_FOUND` (or at minimum drop the `success: true` contradiction)

B. **Marker-side (already tracked in #978):** fix the built marker so it does not enter false-negative states in the first place. Even if #978 lands, layer A is still worth doing as defense-in-depth — consumers should not trust a single boolean for "is the index populated".

**Test gap**

Add a dogfood test that:
1. Builds a small index
2. Manually unsets the built marker (simulating #978)
3. Asserts `--callers <missing-symbol>` does NOT emit the "index empty" hint (only emits it when edges are actually absent)
4. Asserts `success` is `false` (or verdict-aligned) on `NOT_FOUND`

This is exactly the kind of bug the v1.13 postmortem § 1 ("no skip-and-paper-over") warns about — the current message papers over the real condition.

**Priority**

P2 — agents following the current hint actively waste time and may loop. Breaks the AGENTS.md dogfood contract ("agents using TSA as their primary feedback instrument").

**Comment by aimasteracc** [comment]
Resolved by #989 (edge-aware hint, merged). Note: the success-flip half was reclassified won't-fix (#983).
